frontend/services/opengl_glfw/OpenGL_GLFW_Service.cpp

```diff
@@ -792,6 +804,47 @@ void OpenGL_GLFW_Service::glfw_onPathDrop_func(const int path_count, const char*
     this->m_windowEvents.dropped_path_events.push_back(paths_);
 }
 
+void OpenGL_GLFW_Service::create_glfw_mouse_cursor(void) {
+    // (By design, on X11 cursors are user configurable and some cursors may be missing. When a cursor doesn't exist,
+    // GLFW will emit an error which will often be printed by the app, so we temporarily disable error reporting.
+    // Missing cursors will return NULL and  megamol::frontend_resources::WindowManipulation::set_mouse_cursor()
+    // function will use the Arrow cursor instead.)
+    /// See imgui_impl_glfw.cpp
+    for (auto& mouse_cursor_ptr : m_pimpl->mouse_cursors) {
+        mouse_cursor_ptr = nullptr;
+    }
+    if (m_pimpl->mouse_cursors.size() != 9) {
+        return;
+    }
+    // matched to enum ImGuiMouseCursor_ in imgui.h
+    GLFWerrorfun prev_error_callback = ::glfwSetErrorCallback(nullptr);
+    m_pimpl->mouse_cursors[0] = ::glfwCreateStandardCursor(GLFW_ARROW_CURSOR); // ImGuiMouseCursor_Arrow
+    m_pimpl->mouse_cursors[1] = ::glfwCreateStandardCursor(GLFW_IBEAM_CURSOR); // ImGuiMouseCursor_TextInput
+    m_pimpl->mouse_cursors[3] = ::glfwCreateStandardCursor(GLFW_VRESIZE_CURSOR); // ImGuiMouseCursor_ResizeNS
+    m_pimpl->mouse_cursors[4] = ::glfwCreateStandardCursor(GLFW_HRESIZE_CURSOR); // ImGuiMouseCursor_ResizeEW
+    m_pimpl->mouse_cursors[7] = ::glfwCreateStandardCursor(GLFW_HAND_CURSOR);    // ImGuiMouseCursor_Hand
+#if ((GLFW_VERSION_MAJOR * 1000 + GLFW_VERSION_MINOR * 100) >= 3400) // glfw 3.4+
+    m_pimpl->mouse_cursors[2] = ::glfwCreateStandardCursor(GLFW_RESIZE_ALL_CURSOR); // ImGuiMouseCursor_ResizeAll
+    m_pimpl->mouse_cursors[5] = ::glfwCreateStandardCursor(GLFW_RESIZE_NESW_CURSOR); // ImGuiMouseCursor_ResizeNESW
+    m_pimpl->mouse_cursors[6] = ::glfwCreateStandardCursor(GLFW_RESIZE_NWSE_CURSOR); // ImGuiMouseCursor_ResizeNWSE
+    m_pimpl->mouse_cursors[8] = ::glfwCreateStandardCursor(GLFW_NOT_ALLOWED_CURSOR); // ImGuiMouseCursor_NotAllowed
+#else
+    m_pimpl->mouse_cursors[2] = ::glfwCreateStandardCursor(GLFW_ARROW_CURSOR); // ImGuiMouseCursor_ResizeAll
+    m_pimpl->mouse_cursors[5] = ::glfwCreateStandardCursor(GLFW_ARROW_CURSOR); // ImGuiMouseCursor_ResizeNESW
+    m_pimpl->mouse_cursors[6] = ::glfwCreateStandardCursor(GLFW_ARROW_CURSOR); // ImGuiMouseCursor_ResizeNWSE
+    m_pimpl->mouse_cursors[8] = ::glfwCreateStandardCursor(GLFW_ARROW_CURSOR); // ImGuiMouseCursor_NotAllowed
+#endif
+    ::glfwSetErrorCallback(prev_error_callback);
+}
+
+void OpenGL_GLFW_Service::update_glfw_mouse_cursor(const int cursor_id) {
+    if (!m_pimpl->config.windowPlacement.noCursor && (cursor_id < m_pimpl->mouse_cursors.size())) {
```

Comment:
Should we check `cursor_id < 0` here? Because in `GetMouseCursor()` it says `return ImGui::GetMouseCursor(); // ImGuiMouseCursor_None = -1`. We could also change `cursor_id` to `unsigned int` and eliminate this possible error here by making somebody else cast/check their provided cursor id - e.g. `GetMouseCursor()`